src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/PlatformEqtTrace.cs

```diff
@@ -147,6 +175,11 @@ public bool InitializeVerboseTrace(string customLogFile)
         /// <inheritdoc/>
         public bool ShouldTrace(PlatformTraceLevel traceLevel)
         {
+            if (DoNotInitialize)
+            {
+                return false;
+            }
+
             switch (traceLevel)
             {
                 case PlatformTraceLevel.Off:
```

Comment:
The intent of this PR is to prevent initialization of EqtTrace in child app domains when EqtTrace is not enable in parent app domain. 
In the parent app domain, we need to initialize `TraceSource` object to find out whether the EqtTrace is enabled using app.config file. And therefore, we can't prevent instantiating `TraceSource` at least once.